nerfstudio/scripts/downloads/eyeful_tower.py

```diff
@@ -0,0 +1,357 @@
+                frame["k1"] = camera["distortion"][0]
+                frame["k2"] = camera["distortion"][1]
+                frame["k3"] = camera["distortion"][4]
+                frame["k4"] = 0.0
+                frame["p1"] = camera["distortion"][2]
+                frame["p2"] = camera["distortion"][3]
+            elif distortion_model == "Fisheye":
+                # fisheye: [k1, k2, k3, _, _, _, p1, p2]
+                frame["k1"] = camera["distortion"][0]
+                frame["k2"] = camera["distortion"][1]
+                frame["k3"] = camera["distortion"][2]
+                frame["p1"] = camera["distortion"][6]
+                frame["p2"] = camera["distortion"][7]
+            else:
+                raise NotImplementedError("This shouldn't happen")
+
+            T = np.array(camera["T"]).T  # Data stored as column-major
+            T = np.linalg.inv(T)
+            T = T[[2, 0, 1, 3], :]
+            T[:, 1:3] *= -1
+            frame["transform_matrix"] = T.tolist()
+
+            frames.append(frame)
+
+        frames = sorted(frames, key=lambda f: f["file_path"])
+
+        output["frames"] = frames
+        output["train_filenames"] = split_filenames["train"]
+        output["val_filenames"] = split_filenames["test"]
+        return output
+
+    def subsample_nerfstudio_transforms(self, transforms: dict, n: int):
+        target = min(len(transforms["frames"]), n)
+        indices = np.round(np.linspace(0, len(transforms["frames"]) - 1, target)).astype(int)
+
+        frames = []
+        for i in indices:
+            frames.append(transforms["frames"][i])
+
+        output = copy.deepcopy(transforms)
+        output["frames"] = frames
+
+        # Remove the unused files from the splits
+        filenames = {f["file_path"] for f in frames}
+        for key in ["train_filenames", "val_filenames"]:
+            output[key] = sorted(list(set(transforms[key]) & filenames))
+
+        return output
+
+    def download(self, save_dir: Path):
+        if len(self.capture_name) == 0:
+            self.capture_name = ("riverview",)
+            print(
+                f"No capture specified, using {self.capture_name} by default.",
+                "Add `--help` to this command to see all available captures.",
+            )
+
+        if len(self.resolution_name) == 0:
+            self.resolution_name = ("jpeg_2k",)
+            print(
+                f"No resolution specified, using {self.resolution_name} by default.",
+                "Add `--help` to this command to see all available resolutions.",
+            )
+
+        captures = set()
+        for capture in self.capture_name:
+            if capture == "all":
+                captures.update([c for c in eyefultower_downloads if c != "all"])
+            else:
+                captures.add(capture)
+        captures = sorted(captures)
+        if len(captures) == 0:
+            print("WARNING: No EyefulTower captures specified. Nothing will be downloaded.")
```

Comment:
I recommend using CONSOLE.print to make it look a bit cleaner, you can also use colors then if you want.

Reply:
Cool, done.